### ISSUE TITLE
Feature request: return the size of the sdist or the wheel

### ISSUE THREAD

**Issue by agriyakhetarpal** [pypa/build#719]
Hi! I was wondering if it's possible to include the size of the wheels or the sdists built by `build` in bytes/kilobytes, say, in the message that `build` displays after a successful invocation of the build command.

## Motivation

1. It's usually difficult to ascertain what files I have been putting into the wheel with `include_package_data` or into the sdist with `MANIFEST.in` – perhaps I'm copying the wrong files or not putting compiled binaries into the wheel, in error
2. The other alternative to a build frontend that is currently available, i.e., `pip wheel`, does this and returns both the size and also the SHA-256 hash of the wheel

## A workaround for now

To get through this and for further features (i.e, returning a recursive tree of the contents) – I am using https://github.com/hynek/build-and-inspect-python-package/ which uses the `wheel unpack` and the `tar` commands under the hood, but even that doesn't return the size of the sdist or the wheel, I need to compute that manually or through the file system implementation of the OS

## Possible methods for implementation

The `tarfile` and the `zipfile` modules from the standard library are fairly usable and can provide an API for determining the size of a tarball or a `.whl` file. They can be imported in `__main__.py` and further used on `artifact_list`, which I suspect from an overview of the code, is a string containing the name of the files. It can be parsed to construct a `Path` for them, which `zipfile` and `tarfile` can use (unless there is a better way).

## Footnotes

I am happy to do this and write a PR shall this end up being a valid suggestion; would appreciate some pointers if it does, thanks!

**Comment by layday** [comment]
What is the benefit of doing this?

**Comment by layday** [comment]
IIUC, you'd like to be able to use the filesize as a loose indicator for including too few/many files in your dists?

**Comment by agriyakhetarpal** [comment]
Yes, and for parity with `pip wheel` – since it does return the size of the output files, and `pypa/build` will become canon over it someday, and it would be nice to have this :)

**Comment by layday** [comment]
I don't know - I don't think feature parity with `pip wheel` is something we need to worry about, the filesize is a really cure measure of a dist's borkedness, and build is already established and preferred over `pip wheel`.  There are some alternative proposals which would have us generate some metadata in a separate file after a successful build this could possibly fit under, but even then, I think it's too niche.

**Comment by agriyakhetarpal** [comment]
I totally agree with you about the point of having alternative places for metadata after a build succeeds (could you please point me to those proposals? thanks!), but there is also no indication at the moment about whether my sdist is 20 MiB and is bloated up unneccessarily, or if it is 16 KiB as intended. `pip wheel` handles that pretty well, even if it is not the best meausre of the contents of the wheel – it's purely a sanity check. Right now, if I build a platform-specific project, `cibuildwheel` prints out a nice summary that indicates the distributions' size and checksums, but it doesn't cover pure Python projects. Having this in `pypa/build` itself would mean that `cibuildwheel` and other downstream projects could infer it, perhaps through a machine-readable output option.

**Comment by henryiii** [comment]
Cibuildwheel already prints this in a human readable form to your log and actions summary.

**Comment by agriyakhetarpal** [comment]
not for pure Python projects, though ;)

**Comment by henryiii** [comment]
Ah, I see you said that, still, it would not be able to take advantage of this, because it still needs to work with other build frontends.

A machine readable report with the filenames (and the other stuff people ask for, size, time, and hash) would be really nice, though. A tool like cibuildwheel can’t know what files were produced. I think we could use the same redirection trick. Though CMake and compilers and suck likely do not write to Python’s stdout.

**Comment by layday** [comment]
Let's roll this into #198.
